src/server_manager/web_app/app.ts

```diff
@@ -289,319 +306,198 @@ export class App {
+    return {
+      id: localServerId(server),
+      name: this.makeDisplayName(server),
+      isManaged: isManagedServer(server),
+      isSynced: !!server.getName(),
+    };
   }
 
-  // Retrieves the server associated with `displayServer`.
-  private async getServerFromRepository(displayServer: DisplayServer): Promise<server.Server|null> {
-    const apiManagementUrl = displayServer.id;
-    let server: server.Server = null;
-    if (displayServer.isManaged) {
-      if (!!this.digitalOceanRepository) {
-        // Fetch the servers from memory to prevent a leak that happens due to polling when creating
-        // a new object for a server whose creation has been cancelled.
-        const managedServers = await this.digitalOceanRepository.listServers(false);
-        server = managedServers.find(
-            (managedServer) => managedServer.getManagementApiUrl() === apiManagementUrl);
+  private makeDisplayName(server: server.Server): string {
+    let name = server.getName() ?? server.getHostnameForAccessKeys();
+    if (!name) {
+      if (isManagedServer(server)) {
+        // Newly created servers will not have a name.
+        name =
+            this.makeLocalizedServerName((server as server.ManagedServer).getHost().getRegionId());
       }
-    } else {
-      server =
-          this.manualServerRepository.findServer({'apiUrl': apiManagementUrl, 'certSha256': ''});
     }
-    return server;
-  }
-
-  private syncServerCreationToUi(server: server.ManagedServer) {
-    this.serverBeingCreated = server;
-    this.syncDisplayServersToUi();
-    // Show creation progress for new servers only after we have a ManagedServer object,
-    // otherwise the cancel action will not be available.
-    this.showServerCreationProgress();
-    this.waitForManagedServerCreation();
+    return name;
   }
 
-  private getDisplayServerBeingCreated(): DisplayServer {
-    if (!this.serverBeingCreated) {
-      return null;
-    }
-    // Set name to the default server name for this region. Because the server
-    // is still being created, the getName REST API will not yet be available.
-    const regionId = this.serverBeingCreated.getHost().getRegionId();
-    const serverName = this.makeLocalizedServerName(regionId);
-    return {
-      // Use the droplet ID until the API URL is available.
-      id: this.serverBeingCreated.getHost().getHostId(),
-      name: serverName,
-      isManaged: true
-    };
+  private addServer(server: server.Server): void {
+    console.log('Loading server', server);
+    const serverId = localServerId(server);
+    this.idServerMap.set(serverId, server);
+    const serverEntry = this.makeServerListEntry(server);
+    this.appRoot.serverList = this.appRoot.serverList.concat([serverEntry]);
+
+    // Once the server is added to the list, do the rest asynchronously.
+    setTimeout(async () => {
+      // Wait for server config to load, then update the server view and list.
```

Comment:
Maybe we wrap this in a function to make it clear that this is post install logic. 

This looks very specific to one particular flow (progress screen -> server view) so we may be able to extract it from `addServer` in the future.

Reply:
This is post install, but it can happen right after you open the app, in case you closed it right after you started the creation. So it must happen on start too.
Also, this updates the list view to make the icon stop spinning. It's not the server view only.